internal/noop_registry.go

```diff
@@ -0,0 +1,85 @@
+package internal
```

Comment:
Introduces the `NoOpRegistry` struct and methods.

Reply:
First impressions:

✅ this is exactly what we talked about
🤔 This interface still 'smells', because 17 methods is a lot.

I think we should break this interface down - the main question is whether doing it now makes this PR more risky or less risky, and more time consuming or less time consuming.

Here's what I'm thinking

Right now we have:

`*Valid()`
`*Invalid()`
`*Dropped()`

for 5 different types, and it's repetitive.

It looks like each line handler accepts the entire `MetricsRegistry` object, but it only calls the 3 Valid/Invalid/Dropped `Incrementer` specific to one type. For example, `SendSpan` only calls `SpansValid`, `SpansInvalid`, and `SpansDropped`, So I think that gives us something to work with to pass each of the handlers a smaller interface that only has methods for its relevant type.

Reply:
Since the `SuccessTracker` interface has been introduced, I can see that `InternalSDKMetricRegistry` only specifies the internal metric families (e.g. points) involved. The fact that each family has `Valid` ,`Invalid` and `Dropped` versions of the metrics is hidden from `InternalSDKMetricRegistry`. For this reason, I'll click "Resolve conversation."